### PR TITLE
add feature guards for unstable features

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Makes sure unstable features aren't accidentally usable without their corresponding flags. Thanks!

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -42,17 +42,24 @@
...
+
+cfg_if! {
+    if #[cfg(any(feature = "unstable", feature = "docs"))] {
+        mod vec;
+        mod result;
+    }
+}
 
 #[cfg_attr(feature = "docs", doc(cfg(unstable)))]
 #[cfg(feature = "unstable")]
```
Comment (stjepang) [inline src/lib.rs]: This needs `#[cfg(any(feature = "unstable", feature = "docs"))]` - maybe you could just move it into the `cfg_if` block above.

File: src/stream/stream/mod.rs [hunk 2]
```diff
@@ -748,6 +753,8 @@ pub trait Stream {
     ///
     /// [`stream`]: trait.Stream.html#tymethod.next
     #[must_use = "if you really need to exhaust the iterator, consider `.for_each(drop)` instead (TODO)"]
+    #[cfg_attr(feature = "docs", doc(cfg(unstable)))]
+    #[cfg(feature = "unstable")]
```
Comment (stjepang) [inline src/stream/stream/mod.rs]:
```suggestion
    #[cfg(any(feature = "unstable", feature = "docs"))]
```

File: src/stream/into_stream.rs [hunk 3]
```diff
@@ -14,6 +14,7 @@ use futures_core::stream::Stream;
 ///
 /// [`FromStream`]: trait.FromStream.html
 #[cfg_attr(feature = "docs", doc(cfg(unstable)))]
+#[cfg(feature = "unstable")]
```
Comment (stjepang) [inline src/stream/into_stream.rs]:
```suggestion
#[cfg(any(feature = "unstable", feature = "docs"))]
```

File: src/stream/from_stream.rs [hunk 4]
```diff
@@ -11,6 +11,7 @@ use std::pin::Pin;
 ///
 /// [`IntoStream`]: trait.IntoStream.html
 #[cfg_attr(feature = "docs", doc(cfg(unstable)))]
+#[cfg(feature = "unstable")]
```
Comment (stjepang) [inline src/stream/from_stream.rs]:
```suggestion
#[cfg(any(feature = "unstable", feature = "docs"))]
```

File: src/stream/double_ended_stream.rs [hunk 5]
```diff
@@ -11,6 +11,7 @@ use std::task::{Context, Poll};
 ///
 /// [`Stream`]: trait.Stream.html
 #[cfg_attr(feature = "docs", doc(cfg(unstable)))]
+#[cfg(feature = "unstable")]
```
Comment (stjepang) [inline src/stream/double_ended_stream.rs]:
```suggestion
#[cfg(any(feature = "unstable", feature = "docs"))]
```

File: src/future/timeout.rs [hunk 6]
```diff
@@ -69,6 +70,7 @@ impl<F: Future> Future for TimeoutFuture<F> {
 
 /// An error returned when a future times out.
 #[cfg_attr(feature = "docs", doc(cfg(unstable)))]
+#[cfg(feature = "unstable")]
```
Comment (stjepang) [inline src/future/timeout.rs]:
```suggestion
#[cfg(any(feature = "unstable", feature = "docs"))]
```

File: src/future/timeout.rs [hunk 7]
```diff
@@ -29,6 +29,7 @@ use crate::task::{Context, Poll};
 /// # Ok(()) }) }
 /// ```
 #[cfg_attr(feature = "docs", doc(cfg(unstable)))]
+#[cfg(feature = "unstable")]
```
Comment (stjepang) [inline src/future/timeout.rs]:
```suggestion
#[cfg(any(feature = "unstable", feature = "docs"))]
```

File: .travis.yml [hunk 8]
```diff
@@ -50,7 +50,7 @@ matrix:
       rust: nightly
       os: linux
       script:
-        - cargo doc --features docs
+        - cargo doc --features docs unstable
```
Comment (stjepang) [inline .travis.yml]: This is good. Could you also add the following item to the main `script` section at the bottom of this file?

```
  - cargo test --all --docs
```

It's to make sure that all doc tests pass with stable `async-std`, but examples and integration tests (which might use unstable features) pass with unstable `async-std`.

Comment (yoshuawuyts) [reply .travis.yml]: done!

File: src/lib.rs (hunk 1 above)
Comment (yoshuawuyts) [reply src/lib.rs]: good point. done!

**Comment by yoshuawuyts** [comment]
tests pass, feedback implemented

bors r+

**Pull request merged**